provider/azure/azure.go

```diff
@@ -498,3 +510,67 @@ func extractAzureTargets(recordSet *dns.RecordSet) []string {
 	}
 	return []string{}
 }
+
+// extractMetadataFromRecordSet extracts Azure RecordSet metadata into endpoint ProviderSpecific properties.
+func extractMetadataFromRecordSet(ep *endpoint.Endpoint, recordSet *dns.RecordSet) {
+	if recordSet.Properties == nil || recordSet.Properties.Metadata == nil {
+		return
+	}
+	for key, value := range recordSet.Properties.Metadata {
+		if value != nil {
+			ep.WithProviderSpecific(providerSpecificMetadataPrefix+key, *value)
+		}
+	}
+}
+
+// extractMetadataFromEndpoint extracts Azure metadata from endpoint ProviderSpecific properties.
+// Properties with prefix "azure/metadata-" are converted to Azure RecordSet metadata.
+func extractMetadataFromEndpoint(ep *endpoint.Endpoint) map[string]*string {
+	if len(ep.ProviderSpecific) == 0 {
+		return nil
+	}
+	metadata := make(map[string]*string)
+	for _, ps := range ep.ProviderSpecific {
+		if key, ok := strings.CutPrefix(ps.Name, providerSpecificMetadataPrefix); ok {
+			if key != "" && ps.Value != "" {
+				metadata[key] = to.Ptr(ps.Value)
+			}
+		}
+	}
+	if len(metadata) == 0 {
+		return nil
+	}
+	return metadata
+}
+
+// parseAzureTagsAnnotation parses the azure-tags annotation value (key1=value1,key2=value2)
+// and adds individual metadata properties to the endpoint.
+func parseAzureTagsAnnotation(ep *endpoint.Endpoint, tagString string) {
```

Comment:
There is the exact same func for (at least) cloudflare:

https://github.com/kubernetes-sigs/external-dns/blob/02abd2457005341f7942f1777b848035ea74b5d9/provider/cloudflare/cloudflare.go#L732-L743

🤔 Wdyt about moving it to source/annotations and use it in both ?